.github/workflows/build/action.yml

```diff
@@ -0,0 +1,36 @@
+---
+
+name: common build tasks
+inputs:
+  java-version:
+    description: 'Java version'
+    required: true
+  distribution:
+    description: 'Java distribution'
+    required: true
+  command:
+    description: 'Command to execute'
+    default: 'bash'
+    required: false
+  shell:
+    description: 'Default shell'
+    default: 'bash'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up JDK ${{ inputs.java-version }}
+      uses: actions/setup-java@v3
+      with:
+        java-version: ${{ inputs.java-version }}
+        distribution: ${{ inputs.distribution }}
+        cache: 'maven'
+
+    - name: Fetch dependencies and build project
+      run: ./mvnw clean install -DskipTests=true -Dmaven.javadoc.skip=true
+      shell: ${{ inputs.shell }}
+
+    - run: ${{ inputs.command }}
```

Comment:
I'm not sure if `clean` will produce an overhead when running with the unstash build artifact.

What's the reason for using `command` and the step `fetch dependencies` rather than running the given command instead?

Reply:
It shouldn't make any difference with the stash/unstash action in place.
I removed the `fetch dependencies` step.